src/librustc/infer/canonical/mod.rs

```diff
@@ -414,8 +414,17 @@ impl<'cx, 'gcx, 'tcx> InferCtxt<'cx, 'gcx, 'tcx> {
             CanonicalVarKind::PlaceholderConst(
                 ty::PlaceholderConst { universe, name },
             ) => {
-                let _ = (universe, name);
-                unimplemented!() // FIXME(const_generics)
+                let universe_mapped = universe_map(universe);
+                let placeholder_mapped = ty::PlaceholderConst {
+                    universe: universe_mapped,
+                    name,
+                };
+                self.tcx.mk_lazy_const(ty::LazyConst::Evaluated(
+                    ty::Const {
+                        val: ConstValue::Placeholder(placeholder_mapped),
+                        ty: self.tcx.types.err, // FIXME(const_generics)
```

Comment:
Sounds like `PlaceholderConst` should contain the type.

Reply:
Yeah, it probably should. The reason I didn't do it earlier was it ended up propagating `<'tcx>` everywhere and I wanted to make sure it was definitely necessary before going down that route.

Reply:
I attempted propagating `<'tcx>` around when originally adding `CanonicalVarKind::Const` and found it to be impossible because `CanonicalVarInfo`s are allocated in the global arena.

Reply:
Sound like we need to bother @nikomatsakis.

But also, aren't const types supposed to be always be "global"?

Reply:
AFAIK, types are only global after they have been lifted and their dependencies on type parameters and such are taken care of.

Reply:
Type parameters are "global" in this sense, only inference variables aren't.
I'm saying that I don't think we should ever have a constant value in the type system with an *uninferred type*.

Reply:
Hmm. Something feels wrong here indeed. I'm trying too put my finger on what it is. I guess in short I don't like the idea of there being a **type** in the canonical-var-info -- I feel like types should **in the canonicalized value**.

I think what I would expect is that the placeholder doesn't replace the entire `ty::Const`, but rather **just** the `ConstValue`. In this way, it can be instantiated without needing the type. 

(Similarly, in the existential case that appears above, you wouldn't make a "Fresh type variable" to represent its type -- that also seems wrong.)